### DISCUSSION TITLE
VictoryOrgChart

### DISCUSSION THREAD

**Discussion by sibelius** [FormidableLabs/victory#2099] [General]
# Bugs and Questions

### Checklist

* [x] This is not a `victory-native` specific issue. (Issues that only appear in `victory-native` should be opened [here](https://github.com/FormidableLabs/victory-native/issues/new))

* [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/) before asking a question

* [x] I am using the latest version of Victory

* [x] I've searched open issues to make sure I'm not opening a duplicate issue

# Feature Requests

### Checklist

* [x] I've read through the [Docs](https://formidable.com/open-source/victory/docs) and [Guides](https://formidable.com/open-source/victory/guides) to make sure this functionality doesn't already exist

* [x] I've searched open issues to make sure I'm not opening a duplicate issue

### Description

I'd like to have an organization chart.

here are some examples
https://github.com/coreseekdev/react-org-chart (this is built with d3 and react)
this is built with jquery https://github.com/dabeng/OrgChart

Please describe the feature you're requesting in detail.

![image](https://user-images.githubusercontent.com/2005841/47859446-1d098f00-ddcd-11e8-8f04-97111ef13ef2.png)

**Comment by boygirl** [comment]
@sibelius interesting! This is quite different than anything currently implemented in Victory. I will leave it open as a feature, but I have to admit it isn't a high priority at this time.

Things that could help push this feature forward would be:
- more community interest
- api proposals

**Comment by sibelius** [comment]
I think the api could be the same as react-org-chart

```jsx
const props = {
  tree: {
    id: 1,
    person: {
      id: 1,
      avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/spbroma/128.jpg',
      department: '',
      name: 'Imelda Haley',
      title: 'CEO',
      totalReports: 5
    },
    hasChild: true,
    children: []
  },
  loadChildren: d => {
    // this could also just be `return tree.children`
    return Promise.resolve(tree.children)
  },
  lineType: 'curve'
}
```

it'd be cool to enable render a react component in each node

**Comment by boygirl** [comment]
@sibelius 
aside from rendering a react component in each node, are there other differences you would want to see in a Victory version? It might make more sense to PR against react-org-chart

**Comment by sibelius** [comment]
`react-org-chart` is not so React like, they didn't componentize really well

there is no issues enabled, and I think they closed source the newest version

using `victory` use could reuse this chart on react native as well

**Comment by boygirl** [comment]
Ah, I see. I need to think about this more. I think if Victory did take on something similar it would be as a generic tree structure rather than an org chart specifically.

**Comment by sibelius** [comment]
any tree structure would work

I'm trying to understand the victory abstraction so I can try to convert some d3 code to it

**Comment by boygirl** [comment]
Victory only uses a small subset of d3 (shapes, scales, interpolations, and a couple layouts) for things like calculating paths and scaling between data and svg space. Everything else is handled by victory code. There isn't an easy way to convert between the two, unfortunately.

**Comment by urbien** [comment]
+1 
@sibelius maybe it could be helpful to you, we just used the forgotten [victory sunburst chart](https://github.com/FormidableLabs/victory-sunburst) and fixed some bug in it. It also has some packaging issues that we had to workaround for.
Sunburst is a good choice for compact representation of hierarchies, but not all client are familiar with this type of visuals, so classical hierarchical representations would be awesome to see in Victory. We need react-native support so Victory is practically the only choice for versatile charts we have. Thank you Victory!
@boygirl  Perhaps https://js.cytoscape.org/ could be adapted to work in Victory?

**Comment by drowsy-dev** [comment]
+1 Would love to see this in Victory

**Comment by oferRounds** [comment]
+1
